configure

```diff
@@ -63,227 +64,266 @@ FLINT_SOLIB=0
-   echo "     --without-blas       Do not use BLAS (default)"
-   echo "     --with-ntl[=<path>]  Build NTL interface and specify its location (default: /usr/local)"
-   echo "     --without-ntl        Do not build NTL interface (default)"
-   echo "     --extensions=<path>  Specify location of extension modules"
-   echo "     --build=arch-os      Specify architecture/OS combination rather than use values from uname -m/-s"
-   echo "     --enable-shared      Build a shared library (default)"
-   echo "     --disable-shared     Do not build a shared library"
-   echo "     --enable-static      Build a static library (default)"
-   echo "     --disable-static     Do not build a static library"
-   echo "     --single             Faster [non-reentrant if tls or pthread not used] version of library (default)"
-   echo "     --reentrant          Build fully reentrant [with or without tls, with pthread] version of library"
-   echo "     --with-gc=<path>     GC safe build with path to gc"
-   echo "     --enable-pthread     Use pthread (default)"
-   echo "     --disable-pthread    Do not use pthread"
-   echo "     --enable-tls         Use thread-local storage (default)"
-   echo "     --disable-tls        Do not use thread-local storage"
-   echo "     --enable-assert      Enable use of asserts (use for debug builds only)"
-   echo "     --disable-assert     Disable use of asserts (default)"
-   echo "     --enable-cxx         Enable C++ wrapper tests"
-   echo "     --disable-cxx        Disable C++ wrapper tests (default)"
-   echo "     --disable-dependency-tracking Disable gcc automated dependency tracking"
-   echo "     CC=<name>            Use the C compiler with the given name (default: gcc)"
-   echo "     CXX=<name>           Use the C++ compiler with the given name (default: g++)"
-   echo "     AR=<name>            Use the AR library builder with the given name (default: ar)"
-   echo "     LDCONFIG=<name>      Use the given ldconfig tool"
-   echo "     CFLAGS=<flags>       Pass the given flags to the compiler"
-   echo "     CXXFLAGS=<flags>     Pass the given flags to the C++ compiler"
-   echo "     ABI=[32|64]          Tell the compiler to use given ABI (default: empty)"
+    echo "\
+Usage: ./configure [option] [var=value]
+
+Help:
+  -h, --help            Display usage information
+  -V, --version         Display FLINT version
+
+Installation, package and extension paths:
+  --prefix=<path>       Specify path to installation location (default: /usr/local)
+  --with-mpir=<path>    Specify location of MPIR (default: /usr/local)
+  --with-gmp=<path>     Specify location of GMP (default: /usr/local)
+  --with-mpfr=<path>    Specify location of MPFR (default: /usr/local)
+  --with-blas[=<path>]  Use BLAS and specify its location (default: /usr)
+  --with-ntl[=<path>]   Build NTL interface and specify its location (default: /usr/local)
+  --with-gc=<path>      GC safe build with path to gc
```

Comment:
I realize this text was there before and you didn't write it, but perhaps this is an opportunity to improve it nevertheless?

My question for this line: Does this refer to Boehm GC? Then perhaps that could be stated explicitly, as there are many GCs out there? Also, what does "safe build" mean?

Perhaps
```suggestion
  --with-gc=<path>     Specify location of Boehm GC and enable compatibility with it (default: ????)
```

Reply:
Yeah, I have never really used GC other than builtins such as Julia's. I should probably let the person responsible for this line fix it.

Reply:
This was implemented for the Macaulay 2 GC. It is no longer needed as far as I'm aware. We could probably remove the option and the corresponding fmpz model entirely if it weren't for old versions of Macaulay 2 possibly still being used.